src/Lean/Widget/UserWidget.lean

```diff
@@ -145,7 +145,7 @@ def getWidgets (args : Lean.Lsp.Position) : RequestM (RequestTask (GetWidgetsRes
   let doc ← readDoc
   let filemap := doc.meta.text
   let pos := filemap.lspPosToUtf8Pos args
-  withWaitFindSnapAtPos args fun snap => do
+  withWaitFindSnap doc (·.endPos >= pos) (notFoundX := return ⟨∅⟩) fun snap => do
```

Comment:
Could we also do an early lookup at getWidgetSource?

Reply:
We could, but it is not as critical here.  1) We do the getWidgetSource request right after the getWidgets requests has finished.  At that point the snapshot is available.  2) The client caches the javascript on first use.

Reply:
What about the following situation: the getWidgets request resolves and comes back with a source id to the editor. In the meantime the user makes some changes, so the snapshot is gone. Then getWidgetSource is called and has to wait. If it's just the same simple change to the predicate, I'd say let's do it in this PR.

Reply:
It's of course a possible behavior.  I have changed getWidgetSource in the same way.